swatch-contracts/src/main/java/com/redhat/swatch/contract/model/ContractMapper.java

```diff
@@ -61,10 +63,19 @@ public interface ContractMapper {
   ContractMetricEntity metricDtoToMetricEntity(Metric metric);
 
   @Mapping(target = "subscriptionNumber", source = "contract.redHatSubscriptionNumber")
-  @Mapping(target = "vendorProductCode", source = "cloudIdentifiers.productCode")
+  @Mapping(target = "vendorProductCode", source = "cloudIdentifiers")
```

Comment:
Can you add a comment to this, as well as the similar methods for mapRhEntitlementsToContractEntity/extractBillingAccountId, to explain the behavior that the cloudIdentifiers json object gets routed through extractVendorProductCode?  It's not immediately obvious reading through the code that it's happening, and instead looks like we're just mapping a whole "cloudIdentifier" object to "vendorProductCode".